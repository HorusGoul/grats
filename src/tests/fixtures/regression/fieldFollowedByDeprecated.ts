/** @gqlType */
class User {
  /**
   * @gqlField name
   * @deprecated
   */
  graphQLName(): string {
    return "Sup";
  }
}
