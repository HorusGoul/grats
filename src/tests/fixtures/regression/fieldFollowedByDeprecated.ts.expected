-----------------
INPUT
----------------- 
/** @gqlType */
class User {
  /**
   * @gqlField name
   * @deprecated
   */
  graphQLName(): string {
    return "Sup";
  }
}

-----------------
OUTPUT
-----------------
-- SDL --
directive @asyncIterable on FIELD_DEFINITION

directive @propertyName(name: String!) on FIELD_DEFINITION

directive @exported(tsModulePath: String!, functionName: String!, argCount: Int!) on FIELD_DEFINITION

type User {
  name: String @deprecated @propertyName(name: "graphQLName")
}
-- TypeScript --
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
const UserType: GraphQLObjectType = new GraphQLObjectType({
    name: "User",
    fields() {
        return {
            name: {
                deprecationReason: "No longer supported",
                name: "name",
                type: GraphQLString,
                resolve(source, args, context, info) {
                    return typeof source.graphQLName === "function" ? source.graphQLName(source, args, context, info) : source.graphQLName;
                }
            }
        };
    }
});
const schema = new GraphQLSchema({
    types: [UserType]
});
export { schema };
